File: medium/165_Compare_Version_Numbers/main.c
#include <stdio.h>
#include <stdlib.h>

int compareVersion(char* version1, char* version2) {

 while (*version1 || *version2) {
  long long v1 = strtol (version1, &version1, 10);
  long long v2 = strtol (version2, &version2, 10);
  if (v1 < v2) return -1;
  if (v1 > v2) return 1;

  if (*version1) version1++;
  if (*version2) version2++;
 }

 return 0;

}

int main (void) {
 char *str = "123.456.789";
 while (*str) {
  printf("%ld\n", strtol(str, &str, 10));
  printf("%c\n", *str);

  puts("Done");
  if (*str) str++;
 }

 // OUTPUT:
 //  123
 //  .
 //  Done
 //  456
 //  .
 //  Done
 //  789
 //  
 //  Done

 return 0;
}

File: medium/165_Compare_Version_Numbers/main.py
class Solution(object):
    def compareVersion(self, version1, version2):
        """
        :type version1: str
        :type version2: str
        :rtype: int
        """
        n1 = version1.split('.')
        n2 = version2.split('.')
        l1 = len(n1)
        l2 = len(n2)
        if (l1 < l2):
            index = l2
        else: 
            index = l1

        for i in range(index):
            if (i >= l1):
                c1 = 0
            else:
                c1 = int(n1[i])

            if (i >= l2):
                c2 = 0
            else:
                c2 = int(n2[i])

            if (c1 > c2):
                return 1
            elif (c1 < c2):
                return -1

        return 0
        
